### PR TITLE
Rasa data

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
- No need to specify file output format for `rasa data convert` - convert data from "md" to "json" and vice versa
- `rasa data split` stores files in data format of input files. If input data contains mixed formats, training data are stored in "json" format.

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

**Comment by tmbo** [comment]
We can not remove the argument from convert. E.g. you can also convert a data dump from luis to a rasa format.

**Comment by tabergma** [comment]
Thanks for pointing it out 👍. Will update the PR.

File: rasa/cli/data.py [hunk 1]
```diff
@@ -68,8 +70,26 @@ def split_nlu_data(args):
...
     train, test = nlu_data.train_test_split(args.training_fraction)
 
-    train.persist(args.out, filename="training_data.json")
-    test.persist(args.out, filename="test_data.json")
+    train.persist(args.out, filename="training_data.{}".format(fformat))
+    test.persist(args.out, filename="test_data.{}".format(fformat))
+
+
+def get_file_format(resource_name: Text) -> Text:
+    files = list_files(resource_name)
```
Comment (tmbo) [inline rasa/cli/data.py]: there is a helper in `rasa.nlu.training_data.loading` called `_guess_format`, I think we should use the same code everywhere

File: rasa/nlu/training_data/training_data.py [hunk 2]
```diff
@@ -138,7 +138,18 @@ def persist(
             os.makedirs(dir_name)
 
         data_file = os.path.join(dir_name, filename)
-        write_to_file(data_file, self.as_json(indent=2))
+
+        fformat = "md" if filename.endswith(".md") else "json"
```
Comment (tmbo) [inline rasa/nlu/training_data/training_data.py]: I think this function might be called when persisting models to make the training data part of the tar.gz

When persisting in the model, we should always use the same format (`json`).

**Pull request merged**